### DISCUSSION TITLE
Optimizing Minio for fast reads/writes at the expense of reliability

### DISCUSSION THREAD

**Discussion by dv-rastogi** [minio/minio#16542] [Q&A]
Hi!
I plan to use Minio for a case where the object store is ephemeral and requires less-to-no reliability, redundancy, backups & data security/encryption. I wish to optimize read & write performance solely. 
Further note that all my queries are just put and get of unique objects as a whole. I am not overwriting/modifying objects or performing partial reads of the objects.
Are there any resources / specific optimizations (maybe even kernel specific) that could help for such a scenario with Minio? 

Furthermore, is there a in-memory write caching config that minio supports? During our benchmarks, we observed that the minio machine consumed very less system memory. Is there a way that the writes could be cached in memory before being flushed to the disk?

Thank you :).

**Comment by klauspost** [comment]
There isn't too much you can do. The only thing I can think of would be to use a smaller set size, say 4 disks. Set env `MINIO_ERASURE_SET_DRIVE_COUNT=4` before creating the cluster.

You can tweak the [storage class](https://min.io/docs/minio/kubernetes/upstream/operations/concepts/erasure-coding.html#storage-classes), but that will only affect writes.
